fleet/fleet.go

```diff
@@ -131,18 +132,15 @@ type Fleet interface {
 func NewFleet(config Config) (Fleet, error) {
 	var trans http.RoundTripper
 
-	switch config.Endpoint.Scheme {
-	case "unix", "file":
-		if config.Endpoint.Host != "" {
-			// This commonly happens if the user misses the leading slash after the
-			// scheme. For example, "unix://var/run/fleet.sock" would be parsed as
-			// host "var".
-			return nil, maskAnyf(invalidEndpointError, "cannot connect to host %q with scheme %q", config.Endpoint.Host, config.Endpoint.Scheme)
+	tunneling := config.Tunnel.Tunnel != ""
+	// If a tunnel is provided we need to overwrite the http.Transport.Dial function
+	// to use the tunnel
+	if tunneling {
```

Comment:
To mee this is about two different things. You only changed the codeflow here. For me this is now more readable and better understandable because now it is linear and straight forward. 

Towards the other point, you are right. How to fix this? I guess it is what it is. Two places is not that bad. In go you repeat yourself quite often. This is sometimes a desired design decisions based on the golang devs intuitions.